sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/_tests_/cas-supplier-integration.scheduler.e2e-spec.ts

```diff
@@ -366,6 +368,250 @@ describe(describeProcessorRootTest(QueueNames.CASSupplierIntegration), () => {
+    // Assert DB was updated.
+    const updateCASSupplier = await db.casSupplier.findOne({
+      select: {
+        id: true,
+        supplierNumber: true,
+        status: true,
+        lastUpdated: true,
+        supplierAddress: true as unknown,
+        supplierStatus: true,
+        supplierStatusUpdatedOn: true,
+        isValid: true,
+        updatedAt: true,
+        modifier: { id: true },
+      },
+      relations: {
+        modifier: true,
+      },
+      where: {
+        id: savedCASSupplier.id,
+      },
+    });
+    const [submittedAddress] =
+      createSupplierNoSiteResponse.submittedData.SupplierAddress;
+    expect(updateCASSupplier).toEqual({
+      id: savedCASSupplier.id,
+      supplierNumber: createSupplierNoSiteResponse.response.supplierNumber,
+      status: "ACTIVE",
+      lastUpdated: expect.any(Date),
+      supplierAddress: {
+        supplierSiteCode:
+          createSupplierNoSiteResponse.response.supplierSiteCode,
+        addressLine1: submittedAddress.AddressLine1,
+        city: submittedAddress.City,
+        provinceState: submittedAddress.Province,
+        country: submittedAddress.Country,
+        postalCode: submittedAddress.PostalCode,
+        status: "ACTIVE",
+        lastUpdated: expect.any(String),
+      },
+      supplierStatus: SupplierStatus.Verified,
+      supplierStatusUpdatedOn: expect.any(Date),
+      isValid: true,
+      updatedAt: expect.any(Date),
+      modifier: { id: systemUsersService.systemUser.id },
+    });
+    // Ensure updatedAt was updated.
+    expect(updateCASSupplier.updatedAt.getTime()).toBeGreaterThan(
+      referenceDate.getTime(),
+    );
+  });
+
+  it("Should create a new site and update the student CAS supplier when an inactive CAS supplier exists with matching addresses.", async () => {
+    // Arrange
+    // Created a student with same address line 1 and postal code from the expected CAS mocked result.
+    // Postal code has a white space that is expected to be removed.
+    const student = await saveFakeStudent(db.dataSource, undefined, {
+      initialValue: {
+        contactInfo: {
+          address: {
+            addressLine1: "3350 DOUGLAS ST",
+            city: "Victoria",
+            country: "Canada",
+            selectedCountry: COUNTRY_CANADA,
+            provinceState: "BC",
+            postalCode: "V8Z 7X9",
+          },
+        } as ContactInfo,
+      },
+    });
+    const referenceDate = new Date();
+    const savedCASSupplier = await saveFakeCASSupplier(db, { student });
+    // Configure CAS mock to return a result for the GetSupplier
+    // with the same supplier number and address line 1 from the
+    // saved CAS supplier but an inactive status.
+    casServiceMock.getSupplierInfoFromCAS = jest.fn(() =>
+      Promise.resolve(
+        createFakeCASSupplierResponse({
+          initialValues: {
+            status: "INACTIVE", // The status is set to "INACTIVE" to mismatch the address.
```

Comment:
The inactive here is the supplier status and not the site status. the factory needs following update to send the site status. 

1. 
![image](https://github.com/user-attachments/assets/0a589b48-8d27-44f7-9180-67d45668e584)

2.  
![image](https://github.com/user-attachments/assets/5e369f65-9821-4228-9739-79e287833ec7)